### ISSUE TITLE
hv.Violin split, if only one unique value, should raise error

### ISSUE THREAD

**Issue by ahuang11** [holoviz/holoviews#4212]
Currently, there's no warning or errors. The output seems wrong, but not sure where it got "Female" from since I subset only "Male"

```
import numpy as np
import pandas as pd
import holoviews as hv
import hvplot.pandas
hv.extension('bokeh')

import seaborn as sns
sns.set(style="whitegrid")
tips = sns.load_dataset("tips")

tips = pd.concat([
    tips.loc[tips['day'] == 'Thur'],
    tips.loc[tips['day'] == 'Fri'],
    tips.loc[tips['day'] == 'Sat'],
    tips.loc[tips['day'] == 'Sun']
])
tips2 = tips.loc[tips['sex'] == 'Male'].copy()
tips2.hvplot.violin('tip', by=['day', 'sex']).opts(split='sex')
```
![image](https://user-images.githubusercontent.com/15331990/73500634-d9635380-4388-11ea-8118-e2dc17da38d3.png)

**Comment by philippjfr** [comment]
The sex column is a category dtype which stores the available categories independent of the data.

**Comment by philippjfr** [comment]
Fixed as of 1.13.3

<img width="715" alt="Screen Shot 2020-09-02 at 2 44 03 AM" src="https://user-images.githubusercontent.com/1550771/91919410-30862980-ecc6-11ea-9ae5-75991e6810ca.png">
